### DISCUSSION TITLE
How to add Border or Background color to floating line diagnostic ?

### DISCUSSION THREAD

**Discussion by gambhirsharma** [LazyVim/LazyVim#2825] [General]
<img width="766" alt="image" src="https://github.com/LazyVim/LazyVim/assets/69895353/a8db8c75-e3bb-4948-8f30-44da22234679">

Here is the SS, I want to add a border and a different background color to this line floating diagnostics

**Comment by dpetka2001** [comment]
To add borders you could do the following
```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      diagnostics = {
        float = {
          border = "rounded",
        },
      },
    },
  },
```
For changing the background color you would have to refer to your colorscheme's docs and see how to change the following groups `NormalFloat` and `FloatBorder`.

**Reply by macintacos** [reply]
Hiya @dpetka2001 - any ideas how this is supposed to work if you're setting `opts` to a function to override keys? The following setup is giving me errors:

```lua
-- LSP configuration that is different from defaults provided by LazyVim
return {
  "neovim/nvim-lspconfig",
  opts = function()
    local keys = require("lazyvim.plugins.lsp.keymaps").get()

    -- disable keymaps we want to override
    keys[#keys + 1] = { "K", false }

    -- add keymaps
    keys[#keys + 1] = { "gh", vim.lsp.buf.hover, desc = "Hover" }
    keys[#keys + 1] = { "gl", vim.diagnostic.open_float, desc = "Open Diagnostic" }

    -- Borders
    return {
      diagnostics = {
        float = {
          border = "rounded",
        },
      },
    }
  end,
}
```

Error I'm getting:

```
   Error  11:58:44 notify.error lazy.nvim Failed to run `config` for nvim-lspconfig

...share/nvim/lazy/LazyVim/lua/lazyvim/plugins/lsp/init.lua:146: attempt to index field 'inlay_hints' (a nil value)

# stacktrace:
  - /LazyVim/lua/lazyvim/plugins/lsp/init.lua:146 _in_ **config**
  - vim/_editor.lua:0 _in_ **cmd**
  - /persistence.nvim/lua/persistence/init.lua:87 _in_ **load**
  - lua/:1
  - vim/_editor.lua:0 _in_ **cmd**
  - /dashboard-nvim/lua/dashboard/theme/doom.lua:24
```

**Reply by macintacos** [reply]
Okay so, I'm not sure if this breaks anything, but I decided to put the `keys` changes in an `init` function instead of `opts`. Setting `opts` to the normal table works fine, and as far as I can tell, the keymaps are also working fine.

If this is the intended way to do both of these things (edit lspconfig's `opts` while also being able to edit the `keys` in the same file), then perhaps [the documentation](https://www.lazyvim.org/plugins/lsp) needs to be updated? That documentation page doesn't really get into how to do both of these things at the same time in the same config, so might be worth clarifying there (cc @folke).

**Reply by dpetka2001** [reply]
You're overwriting the `nvim-lspconfig` the way you're doing it. Try to do 
```lua
opts = function(_, opts)
  -- keys go here
  opts.diagnostics = {
    float = {
      border = "rounded",
    },
  }
end
```

**Reply by macintacos** [reply]
Ah that works, thank you!

**Comment by noidilin** [comment]
I can't get lsp hover border displaying after neovim v0.11, until I found this [reply](https://github.com/LazyVim/LazyVim/issues/556#issuecomment-1867981476) in #556. It seems like some of UI component have been controlled by noice.nvim.
